gossip/tx_scrambler.go

```diff
@@ -15,8 +19,46 @@ type ScramblerEntry interface {
 	Sender() common.Address
 	// Nonce returns the transaction nonce
 	Nonce() uint64
-	// GasPrice returns the transaction gasPrice price
-	GasPrice() uint64
+	// GasPrice returns the transaction gas price
+	GasPrice() *big.Int
+	// Unwrap returns the wrapped transaction.
+	Unwrap() *types.Transaction
+}
+
+// FilterAndOrderTransactions filters out duplicate transactions and sorts them.
+func FilterAndOrderTransactions(entries []ScramblerEntry) types.Transactions {
+	sorted := filterAndOrderTransactions(entries)
+	txs := make(types.Transactions, len(sorted))
+	for idx, e := range sorted {
+		txs[idx] = e.Unwrap()
+	}
+
+	return txs
+}
+
+// newScramblerTransaction creates a wrapper around *types.Transaction which implements ScramblerEntry.
+func newScramblerTransaction(signer types.Signer, tx *types.Transaction) ScramblerEntry {
```

Comment:
This function actually only requires the sender of the transaction and does not need to check the signature. So it would be enough to just pass the sender.

API functions should always try to minimize what they are asking for. In this case, this just means that the signature check moves to the call side, but in different cases the sender may by available through another mechanism.

Reply:
I could not find any other way to derive the sender from the transaction. I will search again....

Reply:
I added a cache wrapper for the `Signer` developed by @thaarok - I believe that should solve that issue, is that correct?

Reply:
We had a conversation with @jenikd - there seems there is no other option to derive the sender from transaction. Using the cached signer is the correct method.